node_package/src/serverRenderReactComponent.ts

```diff
@@ -83,18 +85,39 @@ Function Component.`);
   }
 
   const consoleReplayScript = buildConsoleReplay();
+  const addRenderingErrors = (resultObject: RenderResult, renderingError: RenderingError) => {
+    resultObject.renderingError = { // eslint-disable-line no-param-reassign
+      message: renderingError.message,
+      stack: renderingError.stack,
+    };
+  }
+
+  if(returnPromise) {
+    const resolveRenderResult = async () => {
+      const promiseResult = {
+        html: await renderResult,
```

Comment:
The entire `promiseResult` gets returned by this async function as a promise, but an `await` statement doesn't seem to resolve internal promises, so I resolve the promise we get from the react component here.